internal/metastore/kv/datacoord/kv_catalog.go

```diff
@@ -247,16 +248,16 @@ func (kc *Catalog) AlterSegmentsAndAddNewSegment(ctx context.Context, segments [
 				return err
 			}
 			maps.Copy(kvs, segmentKvs)
-		} else {
-			// should be a faked segment, we create flush path directly here
-			flushSegKey := buildFlushedSegmentPath(newSegment.GetCollectionID(), newSegment.GetPartitionID(), newSegment.GetID())
-			clonedSegment := proto.Clone(newSegment).(*datapb.SegmentInfo)
-			clonedSegment.IsFake = true
-			segBytes, err := marshalSegmentInfo(clonedSegment)
-			if err != nil {
-				return err
-			}
-			kvs[flushSegKey] = segBytes
+			//} else {
```

Comment:
what if GetNumOfRows ==0 ?

Reply:
will reset it.

Reply:
it is used to write flushed segment, it is no longer necessary.